lib/taskana-core-test/src/test/java/acceptance/classification/delete/DeleteClassificationAccTest.java

```diff
@@ -0,0 +1,240 @@
+
+    assertThatThrownBy(() -> classificationService.getClassification(child.getId()))
+        .isInstanceOf(ClassificationNotFoundException.class);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_DeleteFromAllDomains_When_DeletingMasterClassification() throws Exception {
+    Classification classification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+
+    classificationService.deleteClassification(classification.getKey(), "");
+
+    assertThatThrownBy(
+            () ->
+                classificationService.getClassification(
+                    classification.getKey(), classification.getDomain()))
+        .isInstanceOf(ClassificationNotFoundException.class);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowException_When_DeletingMasterClassificationWithExistingAttachment()
+      throws Exception {
+    Classification classification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Classification attachmentClassification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Attachment attachment =
+        TaskAttachmentBuilder.newAttachment()
+            .classificationSummary(attachmentClassification.asSummary())
+            .objectReference(DefaultTestEntities.defaultTestObjectReference().build())
+            .build();
+    TaskBuilder.newTask()
+        .classificationSummary(classification.asSummary())
+        .workbasketSummary(defaultWorkbasket)
+        .attachments(attachment)
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(
+            () -> classificationService.deleteClassification(attachmentClassification.getId()))
+        .isInstanceOf(ClassificationInUseException.class);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_ThrowExceptionAndRollback_When_ChildClassificationIsInUse() throws Exception {
+    Classification parent =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    Classification child =
+        DefaultTestEntities.defaultTestClassification()
+            .parentId(parent.getId())
+            .buildAndStore(classificationService);
+    TaskBuilder.newTask()
+        .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+        .classificationSummary(child)
+        .workbasketSummary(defaultWorkbasket)
+        .buildAndStore(taskService);
+
+    assertThatThrownBy(() -> classificationService.deleteClassification(parent.getId()))
+        .isInstanceOf(ClassificationInUseException.class);
+    Classification rollback =
+        classificationService.getClassification(parent.getKey(), parent.getDomain());
+    assertThat(rollback.getDomain()).isEqualTo(parent.getDomain());
+  }
```

Comment:
I think it would be better to use: `assertThat(rollback).isEqualTo(parent);`

Reply:
Good idea!